### ISSUE TITLE
Inaccurate Assignment of Gene ID and Transcript ID on Intron Feature of Tiberius Annotation

### ISSUE THREAD

**Issue by xo2003** [Gaius-Augustus/Tiberius#4]
Hi,
I recently ran Tiberius annotation on one of our insect genomes, utilizing the soft-masking model weights under CPU. I found that Tiberius performs faster than BRAKER3 in CPU mode, but the gene number is approximately 2,500 fewer than what BRAKER3 produced. I suspect this discrepancy may be due to the training model not being suitable for invertebrate genomes currently.

My main issue is that Tiberius did not assign both `gene_id` and `transcript_id` correctly for the intron features. I have attached a cropped figure of the GTF for your reference. Hope this can be addressed in future updates. Thank you!

![image](https://github.com/user-attachments/assets/654fc043-5b93-4579-89e9-95e74a01dce2)
![image](https://github.com/user-attachments/assets/0ae8f98f-956f-4677-a511-d41ebcaed745)

**Comment by LarsGab** [comment]
Hi,

thanks for testing TIberius. I agree that the lower gene count is due to the model being trained on mammalian genomes. I would also expect it to be less accurate on insects than mammals until we retrain it for insects. See Figure 4 in our [preprint](https://doi.org/10.1101/2024.07.21.604459) or the last second figure in the readme for a benchmark on non-mammalian genomes. 

I fixed the bug with the IDs of the intron lines in the GTF.

Best,
Lars

**Comment by xo2003** [comment]
Hi Lars,

Thank you for your prompt reply! Yes, I noticed that the current model is optimized for mammalian genomes, and I saw the non-mammalian performance in Figure 4 of the preprint. That’s why I didn’t initially run Tiberius. As you mentioned in the paper, deep learning-based annotation offers a solution for species with insufficient genomic evidence. We’re encountering similar challenges with other vertebrate species, and I truly appreciate the approach your team has developed.

I attempted to run Tiberius in GPU mode on several machines, but regular computers couldn’t handle the job—likely due to insufficient memory, which caused the jobs to terminate. The annotation only completed successfully on an AI station with an A100 40GB 2-pin GPU, and a small genome (~120 Mbp) finished in about 10 minutes. Could you share what the minimum GPU requirements are?

Thanks again for fixing the bug!

**Comment by LarsGab** [comment]
Hi,

you can run Tiberius on a GPU with a smaller memory by reducing the batch size. You can use the argument `--batch_size` to adjust it. The default batch size is 16, but I was able to run Tiberius with a batch size of 8 on a RTX 3090 (25GB memory) and with a batch size of 4 on a RTX 2070 (8GB memory).

Best,
Lars

**Comment by xiekunwhy** [comment]
> Hi,
> 
> thanks for testing TIberius. I agree that the lower gene count is due to the model being trained on mammalian genomes. I would also expect it to be less accurate on insects than mammals until we retrain it for insects. See Figure 4 in our [preprint](https://doi.org/10.1101/2024.07.21.604459) or the last second figure in the readme for a benchmark on non-mammalian genomes.
> 
> I fixed the bug with the IDs of the intron lines in the GTF.
> 
> Best, Lars

Hi, @LarsGab , I find "until we retrain it for insects" in your reply, do you mean that you are trainning some other non-mammalian model, and will release weight files when trainning is finish?

**Comment by MarioStanke** [comment]
Yes, someone in our group is training it on insects, broadly and in narrower clades (among them Lepidoptera).
Don't expect the parameter release in less than a few months. We will have to make careful experiments to see which training works best.

**Comment by xiekunwhy** [comment]
> Yes, someone in our group is training it on insects, broadly and in narrower clades (among them Lepidoptera). Don't expect the parameter release in less than a few months. We will have to make careful experiments to see which training works best.

Any propgress ? Anxiously awaiting for other models because we have no enough resource to train a new model.

Best,
Kun
